plenoptic/metric/perceptual_distance.py

```diff
@@ -399,62 +415,86 @@ def nlpd(IM_1, IM_2):
 
     Parameters
     ----------
-    IM_1: torch.Tensor
-        image, (1 x 1 x H x W)
-    IM_2: torch.Tensor
-        image, (1 x 1 x H x W)
-
+    IM_1: torch.Tensor of shape (batch, channel, height, width)
+        The first image or batch of images. Channels are also treated as batches.
+    IM_2: torch.Tensor of shape (batch, channel, height, width)
+        The second image or batch of images. Channels are also treated as batches.
     Returns
     -------
-    distance: float
-
-    Note
-    ----
-    only accepts single channel images
+    distance: torch.Tensor of shape (batch, channel)
+        The normalized Laplacian Pyramid distance.
 
     References
     ----------
     .. [1] Laparra, V., Ballé, J., Berardino, A. and Simoncelli, E.P., 2016. Perceptual image quality
        assessment using a normalized Laplacian pyramid. Electronic Imaging, 2016(16), pp.1-6.
     """
 
-    y = normalized_laplacian_pyramid(torch.cat((IM_1, IM_2), 0))
+    if IM_1.shape[1] > 1 or IM_2.shape[1] > 1:
+        warnings.warn("NLPD was developed on grayscale images, so the channel dimension "
+                      "is treated as another batch dimension.")
+    y1 = normalized_laplacian_pyramid(IM_1)
+    y2 = normalized_laplacian_pyramid(IM_2)
 
-    # for optimization purpose (stabilizing the gradient around zero)
-    epsilon = 1e-10
+    epsilon = 1e-10  # for optimization purpose (stabilizing the gradient around zero)
     dist = []
     for i in range(6):
-        dist.append(torch.sqrt(torch.mean((y[i][0] - y[i][1]) ** 2) + epsilon))
+        dist.append(torch.sqrt(torch.mean((y1[i] - y2[i]) ** 2, dim=(2, 3)) + epsilon))
 
-    return torch.stack(dist).mean()
+    return torch.stack(dist).mean(dim=0)
 
 
-def nspd(IM_1, IM_2, O=1, S=5, complex=True):
+def normalized_steerable_pyramid(im, sigmas=None):
+    """Compute the normalized Steerable Pyramid using pre-optimized parameters. (under construction)
+
+    Arguments
+    --------
+    im: torch.Tensor of shape (batch, channel, height, width)
+        Image, or batch of images. Channels are also treated as batches.
```

Comment:
Wording of "Channels are also treated as batches" strikes me as a bit unclear. I'd say something like "computed independently for each batch and channel" (because that's what it does, right?)

Reply:
Yes that's better.